### DISCUSSION TITLE
External link is not working in Next.js when you want to use Link component

### DISCUSSION THREAD

**Discussion by olavomello** [vercel/next.js#23286] [Help] [answered]
Anyone know this problem?
My external JS doesn't work when content is showed by using Link.
Tkz ;)

**Comment by akellbl4** [comment, marked as answer]
Could you please share an example of how you are trying to use `Link`?
What you mean under `not working`? Any errors or the link just doesn't open?

**Reply by olavomello** [reply]
Hi @akellbl4 

sorry to not be clear about my problem. Calling .js in a specific page and accessing the route throught component LINK, the scripts don't be executed. If I access route by URL everything is ok. Can you understand?

Thanks for help

**Reply by akellbl4** [reply]
Okay, If I'm right you have some injected JS in your page, and if you open it directly everything works.
Is your JS inlined or you load it from a file?

**Reply by olavomello** [reply]
Exactly

**Reply by akellbl4** [reply]
Could you please share an example of how your peace of JS is called in JSX? After that, I could propose some solution for that.

**Reply by olavomello** [reply]
Sure and tkz again for help.

Page : 
       ...
      <script src="/js/chart-area.js"></script> 
      <script src="/js/chart-pie.js"></script> 
    </ContentWrapper>
    
  Called by :
        ....
        <Link passHref={true} href="/">
          <a className="nav-link">
            <i className="fas fa-fw fa-tachometer-alt"></i>
            <span>Dashboard</span>
          </a>
        </Link>
        
        
        
Accessing the route directly from URL, as an A HREF action, the "chart script .js" is normally called but using LINK ( that have a function as an AJAX ) the "chart script .js" isn't called.   I fill that using LINK I need to apply some type of identifier to "scripts" called in content. Exactly that same problem that we have using calls in js AJAX content. 

Tkz bro

**Reply by akellbl4** [reply]
I have two solutions for you

The first one is to load JS dynamically after page render.

```jsx
function loadScript(url, onLoad, onError) {
  const script = document.createElement("script")
  script.src = url
  script.onload = onLoad
  script.onerror = onError
  document.body.appendChild(script)
}

function Page() {
  useEffect(() => {
    loadScript('/js/chart-area.js')
    loadScript('/js/chart-pie.js')
  }, [])

  return <div>{...}</div>
}
```
And the same load with modern `import` (I haven't tried to load scripts like that in a Next.js app)

```js
function loadScript(url, onLoad, onError) {
  import(url).then(onLoad).catch(onError)
}
```

The second one to import those scripts as modules (for this example you should set the right path to files in your project)

```jsx
import 'scripts/chart-area.js'
import 'scripts/chart-pie.js'

function Page() {
  return <div>{...}</div>
}
```

If your scrips are trying to access DOM or browser-specific API the first option could be better.

**Reply by olavomello** [reply]
Dude thank you so much. I´ll try !

**Reply by akellbl4** [reply]
Mark the thread as answered in case the solution is working or write down what's wrong and I'll try to help ;)

**Reply by olavomello** [reply]
Problem solved @akellbl4 . Thank you so much !
